### PR TITLE
[WIP]Merge tickets

### PR DESCRIPTION
**Pull request by hin101**
Functionality done styling left

### CODE REVIEW

**Comment by hin101** [comment]
Issue #441

**Comment by CGA1123** [comment]
Adding i18n strings to this would be ideal!

File: app/controllers/admin/topics_controller.rb [hunk 1]
```diff
@@ -366,6 +366,49 @@ def split_topic
...
+          @posts.create(
+          body: p.body,
+          kind: p.kind,
+          user: p.user,
+          screenshots: p.screenshots,
+          attachments: p.attachments,
+          )
+        end
+
+        topics_merged << "%#{t.id}-%#{t.name}\n"
```
Comment (scott) [inline app/controllers/admin/topics_controller.rb]: You don't actually need the % signs, that was actually part of the token in the example I gave.

File: app/views/admin/topics/_tickets.html.erb [hunk 2]
```diff
@@ -68,6 +68,7 @@
           <% end %>
 
         </span>
+        <span class="merge-span" style="display: none;"><%= link_to t(:merge, default: "Merge"), admin_merge_tickets_path(merge:''), remote: true, class: 'multiple-update merge', data: { confirm: "Are you sure you want to merge these tickets?" } %></span>
```
Comment (scott) [inline app/views/admin/topics/_tickets.html.erb]: We will need i18n for the confirm message too

File: app/assets/stylesheets/admin.scss [hunk 3]
```diff
@@ -650,3 +650,9 @@ ul.settings-menu {
     }
   }
 }
+
+.merge {
+  color: red;
+  font-weight: bold;
+  text-transform: uppercase;
+}
```
Comment (scott) [inline app/assets/stylesheets/admin.scss]: We don't actually need the red, I should have noted in the requirements that it was red just to call attention to the new link.  So in other words, the merge link should look like the others in that menu bar.

File: app/controllers/admin/topics_controller.rb [hunk 4]
```diff
@@ -366,6 +366,49 @@ def split_topic
...
+        logger.info("Id #{t.id}")
+        t.posts.each do |p|
+          @posts.create(
+          body: p.body,
+          kind: p.kind,
+          user: p.user,
+          screenshots: p.screenshots,
+          attachments: p.attachments,
+          )
+        end
```
Comment (scott) [inline app/controllers/admin/topics_controller.rb]: One issue that I didn't think about in the requirements is that this generates two `Posts` with a `kind: 'first'` which causes problems with the formatting.  In this case, we should change the kind to `reply` for all `Posts` except that first one belonging to  `Topic.first`.  Hope that makes sense.

**Comment by scott** [comment]
When I look at the logs for this I am seeing a weird request- `Started GET "/admin/topics/merge_tickets?locale=en&merge=&topic_ids[]=65&topic_ids[]=98&topic_ids[]=65&topic_ids[]=98"`

Should the `topics_ids[]` appear twice?  Also the `merge` param appears to have no value?

File: app/controllers/admin/topics_controller.rb (hunk 4 above)
Comment (CGA1123) [inline app/controllers/admin/topics_controller.rb]: Maybe we should add a validation to check for a single post of kind `first` scoped to topic?

**Comment by CGA1123** [comment]
Should this feature really be using a `GET` request? Would `PUT` or `PATCH` not be more appropriate?

**Comment by scott** [comment]
@CGA1123 The thought crossed my mind as well, I think it should be a `PUT` since it is creating a new ticket from the other two.

File: app/assets/javascripts/app.js [hunk 5]
```diff
@@ -224,17 +224,24 @@ Helpy.ready = function(){
     var output;
     var messages = $('.topic-checkbox:checked').size();
 
+    $('.merge-span').hide();
+    console.log("Update");
+
     switch(messages) {
       case 1:
         output = Helpy.selected[1];
+        $('.merge-span').hide();
```
Comment (hin101) [inline app/assets/javascripts/app.js]: This is so the merge link only shows when two or more tickets are selected.

Comment (scott) [inline app/assets/javascripts/app.js]: Oh okay, good thinking :)

**Comment by hin101** [comment]
I put it as a get request as I want to redirect back to the new topic

**Pull request merged**